core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -169,8 +168,8 @@
     private final Map<DynamicFilterId, DynamicFilterDomainStats> dynamicFilterDomainStats;
     private final ValuePrinter valuePrinter;
 
-    // NOTE: do NOT add Metadata or Session to this class.  The plan printer must be usable outside of a transaction.
-    private PlanPrinter(
+    // NOTE: do NOT add Metadata or Session to this class.  The plan printer must be usable outside a transaction.
+    public PlanPrinter(
```

Comment:
Keep this constructor private and remove the call to create an instance from within the JsonRenderer. The relationship should go the other way. PlanPrinter should rely on JsonRenderer to render a plan as a json string.